### ISSUE TITLE
Release 4.4.0 - Release Candidate 1 - External Integrations modules

### ISSUE THREAD

**Issue by fdalmaup** [wazuh/wazuh#16184]
The following issue aims to perform all the required testing for the current release candidate to ensure the modules (AWS, GCP, Azure, DockerListener and Shuffle) works as expected, report the results, and open new issues for any encountered errors.

## Modules tests information

|||
|----------------------------------|------ |
| **Main release candidate issue** | #16132    |
| **Version**                      | 4.4.0 |
| **Release candidate #**          |   RC1  |
| **Tag**                          | [v4.4.0-rc1](https://github.com/wazuh/wazuh/tree/v4.4.0-rc1) |
| **Previous modules tests issue** | #15750    |

## Test report procedure

All test results must have one of the following statuses: 
|                                  |                                                        |
|-----------------|-------------------------------------------------------------------------|
| :green_circle:  | The test passed successfully. 										    |
| :yellow_circle: | The test passed but some enhancements need to be applied to the module. |
| :red_circle:    | The test failed.                                                        |
| :white_circle:  | The test does not apply.                                                |

Any failing test must be properly addressed with a new issue, detailing the error and the possible cause. It must be included in the `Fixes` section of the current release candidate main issue.

The resulting logs for the tests must be included in the status report so it can be used by the auditors to dig deeper into any possible failures and details.

## Conclusions

| **Module**      |  **Status**  |**Notes**|
|-----------------|-------------|---------|
| AWS             | :green_circle:  |  |
| Azure           |  :green_circle:   |  |
| GCP             | :green_circle:  |   |
| Docker Listener |  :green_circle:  |  |
| Shuffle |  :green_circle:  |  |

## Auditors validation
The definition of done for this one is the validation of the conclusions and the test results from all auditors.

All checks from below must be accepted in order to close this issue.
- [ ] @davidjiglesias 
- [x] @fdalmaup

**Comment by fdalmaup** [comment]
# GCP Test results

No data must be modified or removed unless it was specified for the test.

The tests assume there is a valid authentication file located at `/var/ossec/wodles/gcloud/credentials.json`.

A test is considered valid if the following conditions are met:

- The expected result described in the test itself is met.
- The expected amount of events were sent to `analysisd`.
- The expected output was present in the `{wazuh_path}/logs/ossec.log` file.

## Test results

| Bucket/Service type            | Status                  | Issue ref. |
|--------------------------------|-------------------------|------------|
| Test using ossec.conf in a manager | :green_circle:  | -          |
| Test using ossec.conf in a agent (Python 3.6)   | :green_circle:  | -          |

## Environment details

To test Pub/Sub integration we developed a script able to publish any number of messages in the testing topic so we can ensure there always will be data available for the tests. This was required because of how Pub/Sub is designed as the messages are consumed once they are pulled.


## Test using ossec.conf in a manager

Run the module using the following `ossec.conf` configuration:

<details><summary>ossec.conf</summary>


    <gcp-pubsub>
        <pull_on_start>yes</pull_on_start>
        <interval>1d</interval>
        <project_id>wazuh-dev-123456</project_id>
        <subscription_name>wazuh-framework-subscription</subscription_name>
        <num_threads>4</num_threads>
        <logging>info</logging>
        <max_messages>50</max_messages>
        <credentials_file>/var/ossec/wodles/gcloud/credentials.json</credentials_file>
    </gcp-pubsub>

    <gcp-bucket>
        <run_on_start>yes</run_on_start>
        <interval>1d</interval>
        <logging>info</logging>
        <bucket type="access_logs">
            <name>framework-test-bucket</name>
            <credentials_file>/var/ossec/wodles/gcloud/credentials.json</credentials_file>
        </bucket>

        <bucket type="access_logs">
            <name>framework-test-bucket</name>
            <credentials_file>/var/ossec/wodles/gcloud/credentials.json</credentials_file>
            <only_logs_after>2022-Jul-01</only_logs_after>
            <path>test_prefix</path>
        </bucket>
    </gcp-bucket>

</details>

<details>
    <summary>Ossec.log output</summary>

    2023/02/13 11:49:09 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:131 at wm_gcp_pubsub_main(): INFO: Module started.
    2023/02/13 11:49:09 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:147 at wm_gcp_pubsub_main(): DEBUG: Starting fetching of logs.
    2023/02/13 11:49:09 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:248 at wm_gcp_pubsub_run(): DEBUG: Create argument list
    2023/02/13 11:49:09 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:301 at wm_gcp_pubsub_run(): DEBUG: Launching command: wodles/gcloud/gcloud --integration_type pubsub --project wazuh-dev-123456 --subscription_id wazuh-framework-subscription --credentials_file /var/ossec/wodles/gcloud/credentials.json --max_messages 50 --num_threads 4 --log_level 2
    2023/02/13 11:49:09 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:170 at wm_gcp_bucket_main(): INFO: Module started.
    2023/02/13 11:49:09 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:188 at wm_gcp_bucket_main(): DEBUG: Starting fetching of logs.
    2023/02/13 11:49:09 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:220 at wm_gcp_bucket_main(): INFO: Executing Bucket Analysis: (Bucket: framework-test-bucket, Type: access_logs, Credentials file: /var/ossec/wodles/gcloud/credentials.json)
    2023/02/13 11:49:09 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:381 at wm_gcp_bucket_run(): DEBUG: Create argument list
    2023/02/13 11:49:09 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:424 at wm_gcp_bucket_run(): DEBUG: Launching command: wodles/gcloud/gcloud --integration_type access_logs --bucket_name framework-test-bucket --credentials_file /var/ossec/wodles/gcloud/credentials.json --log_level 2
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_1_new.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:346 at wm_gcp_parse_output(): DEBUG: Setting 4 threads to pull 50 messages in total
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_22.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_2_new.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_3.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:346 at wm_gcp_parse_output(): DEBUG: Processing event: {"integration": "gcp", "gcp": [Mon Feb 13 11:48:09 2023] - Framework publish message 8}
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_3_new.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_4.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_5.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:346 at wm_gcp_parse_output(): DEBUG: Sending msg to analysisd: "b'1:Wazuh-GCloud:{"integration": "gcp", "gcp": [Mon Feb 13 11:48:09 2023] - Framework publish message 8}'"
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_6.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_7.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_8.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_9.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:346 at wm_gcp_parse_output(): DEBUG: Processing event: {"integration": "gcp", "gcp": [Mon Feb 13 11:48:09 2023] - Framework publish message 10}
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_release_4_4.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of testingfile.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: Received 0 messages
    2023/02/13 11:49:13 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:346 at wm_gcp_parse_output(): DEBUG: Sending msg to analysisd: "b'1:Wazuh-GCloud:{"integration": "gcp", "gcp": [Mon Feb 13 11:48:09 2023] - Framework publish message 10}'"
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:220 at wm_gcp_bucket_main(): INFO: Executing Bucket Analysis: (Bucket: framework-test-bucket, Path: test_prefix, Type: access_logs, Credentials file: /var/ossec/wodles/gcloud/credentials.json)
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:381 at wm_gcp_bucket_run(): DEBUG: Create argument list
    2023/02/13 11:49:13 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:424 at wm_gcp_bucket_run(): DEBUG: Launching command: wodles/gcloud/gcloud --integration_type access_logs --bucket_name framework-test-bucket --credentials_file /var/ossec/wodles/gcloud/credentials.json --prefix test_prefix --only_logs_after 2022-Jul-01 --log_level 2
    2023/02/13 11:49:13 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:346 at wm_gcp_parse_output(): DEBUG: Processing event: {"integration": "gcp", "gcp": [Mon Feb 13 11:48:08 2023] - Framework publish message 0}
    2023/02/13 11:49:13 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:346 at wm_gcp_parse_output(): DEBUG: Sending msg to analysisd: "b'1:Wazuh-GCloud:{"integration": "gcp", "gcp": [Mon Feb 13 11:48:08 2023] - Framework publish message 0}'"
    ...
    2023/02/13 11:49:13 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: Received and acknowledged 50 messages
    2023/02/13 11:49:13 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:151 at wm_gcp_pubsub_main(): DEBUG: Fetching logs finished.
    2023/02/13 11:49:13 wazuh-modulesd:gcp-pubsub[16452] wm_gcp.c:143 at wm_gcp_pubsub_main(): DEBUG: Sleeping until: 2023/02/14 11:49:09
    2023/02/13 11:49:14 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_prefix/test_1.txt is older than 2022-07-01 00:00:00+00:00. Skipping it...
    2023/02/13 11:49:14 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: Processing test_prefix/test_2.txt
    2023/02/13 11:49:14 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:346 at wm_gcp_parse_output(): DEBUG: Sending msg to analysisd: "b'1:Wazuh-GCloud:{"integration": "gcp", "gcp": {"GOOGLE CLOUD STORAGE TEST": "Log 1", "source": "gcp_bucket"}}'"
    2023/02/13 11:49:14 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: Updating previously processed files.
    2023/02/13 11:49:14 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: Received 1 message
    2023/02/13 11:49:14 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:225 at wm_gcp_bucket_main(): DEBUG: Fetching logs finished.
    2023/02/13 11:49:14 wazuh-modulesd:gcp-bucket[16452] wm_gcp.c:184 at wm_gcp_bucket_main(): DEBUG: Sleeping until: 2023/02/14 11:49:09
  
</details>

### Conclusions
The module worked as expected, 50 events were sent to `analysisd`.


## Test using ossec.conf in an agent with Python 3.6 

Run the module using the following `ossec.conf` configuration:

<details><summary>ossec.conf</summary>


    <gcp-pubsub>
        <pull_on_start>yes</pull_on_start>
        <interval>1d</interval>
        <project_id>wazuh-dev-123456</project_id>
        <subscription_name>wazuh-framework-subscription</subscription_name>
        <num_threads>4</num_threads>
        <logging>info</logging>
        <max_messages>50</max_messages>
        <credentials_file>/var/ossec/wodles/gcloud/credentials.json</credentials_file>
    </gcp-pubsub>

    <gcp-bucket>
        <run_on_start>yes</run_on_start>
        <interval>1d</interval>
        <logging>info</logging>
        <bucket type="access_logs">
            <name>framework-test-bucket</name>
            <credentials_file>/var/ossec/wodles/gcloud/credentials.json</credentials_file>
        </bucket>

        <bucket type="access_logs">
            <name>framework-test-bucket</name>
            <credentials_file>/var/ossec/wodles/gcloud/credentials.json</credentials_file>
            <only_logs_after>2022-Jul-01</only_logs_after>
            <path>test_prefix</path>
        </bucket>
    </gcp-bucket>


</details>

<details>
    <summary>Ossec.log output</summary>

    2023/02/13 15:01:54 wazuh-modulesd:gcp-pubsub[10243] wm_gcp.c:131 at wm_gcp_pubsub_main(): INFO: Module started.
    2023/02/13 15:01:54 wazuh-modulesd:gcp-pubsub[10243] wm_gcp.c:147 at wm_gcp_pubsub_main(): DEBUG: Starting fetching of logs.
    2023/02/13 15:01:54 wazuh-modulesd:gcp-pubsub[10243] wm_gcp.c:248 at wm_gcp_pubsub_run(): DEBUG: Create argument list
    2023/02/13 15:01:54 wazuh-modulesd:gcp-pubsub[10243] wm_gcp.c:301 at wm_gcp_pubsub_run(): DEBUG: Launching command: wodles/gcloud/gcloud --integration_type pubsub --project wazuh-dev-123456 --subscription_id wazuh-framework-subscription --credentials_file /var/ossec/wodles/gcloud/credentials.json --max_messages 50 --num_threads 4 --log_level 2
    2023/02/13 15:01:54 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:170 at wm_gcp_bucket_main(): INFO: Module started.
    2023/02/13 15:01:54 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:188 at wm_gcp_bucket_main(): DEBUG: Starting fetching of logs.
    2023/02/13 15:01:54 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:220 at wm_gcp_bucket_main(): INFO: Executing Bucket Analysis: (Bucket: framework-test-bucket, Type: access_logs, Credentials file: /var/ossec/wodles/gcloud/credentials.json)
    2023/02/13 15:01:54 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:381 at wm_gcp_bucket_run(): DEBUG: Create argument list
    2023/02/13 15:01:54 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:424 at wm_gcp_bucket_run(): DEBUG: Launching command: wodles/gcloud/gcloud --integration_type access_logs --bucket_name framework-test-bucket --credentials_file /var/ossec/wodles/gcloud/credentials.json --log_level 2
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_1_new.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_22.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_2_new.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_3.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_3_new.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_4.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_5.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_6.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_7.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_8.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_9.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_release_4_4.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of testingfile.txt is older than 2023-02-13 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: Received 0 messages
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:220 at wm_gcp_bucket_main(): INFO: Executing Bucket Analysis: (Bucket: framework-test-bucket, Path: test_prefix, Type: access_logs, Credentials file: /var/ossec/wodles/gcloud/credentials.json)
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:381 at wm_gcp_bucket_run(): DEBUG: Create argument list
    2023/02/13 15:01:55 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:424 at wm_gcp_bucket_run(): DEBUG: Launching command: wodles/gcloud/gcloud --integration_type access_logs --bucket_name framework-test-bucket --credentials_file /var/ossec/wodles/gcloud/credentials.json --prefix test_prefix --only_logs_after 2022-Jul-01 --log_level 2
    2023/02/13 15:01:56 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: The creation time of test_prefix/test_1.txt is older than 2022-07-01 00:00:00+00:00. Skipping it...
    2023/02/13 15:01:56 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: Processing test_prefix/test_2.txt
    2023/02/13 15:01:56 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:346 at wm_gcp_parse_output(): DEBUG: Sending msg to analysisd: "b'1:Wazuh-GCloud:{"integration": "gcp", "gcp": {"GOOGLE CLOUD STORAGE TEST": "Log 1", "source": "gcp_bucket"}}'"
    2023/02/13 15:01:56 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: Updating previously processed files.
    2023/02/13 15:01:56 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: Received 1 message
    2023/02/13 15:01:56 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:225 at wm_gcp_bucket_main(): DEBUG: Fetching logs finished.
    2023/02/13 15:01:56 wazuh-modulesd:gcp-bucket[10243] wm_gcp.c:184 at wm_gcp_bucket_main(): DEBUG: Sleeping until: 2023/02/14 15:01:54
    2023/02/13 15:01:57 wazuh-modulesd:gcp-pubsub[10243] wm_gcp.c:346 at wm_gcp_parse_output(): DEBUG: Setting 4 threads to pull 50 messages in total
    2023/02/13 15:01:57 wazuh-modulesd:gcp-pubsub[10243] wm_gcp.c:346 at wm_gcp_parse_output(): DEBUG: Processing event: {"integration": "gcp", "gcp": [Mon Feb 13 11:54:38 2023] - Framework publish message 6}
    2023/02/13 15:01:57 wazuh-modulesd:gcp-pubsub[10243] wm_gcp.c:346 at wm_gcp_parse_output(): DEBUG: Sending msg to analysisd: "b'1:Wazuh-GCloud:{"integration": "gcp", "gcp": [Mon Feb 13 11:54:38 2023] - Framework publish message 6}'"
    2023/02/13 15:01:57 wazuh-modulesd:gcp-pubsub[10243] wm_gcp.c:352 at wm_gcp_parse_output(): INFO: Received and acknowledged 50 messages
    2023/02/13 15:01:57 wazuh-modulesd:gcp-pubsub[10243] wm_gcp.c:151 at wm_gcp_pubsub_main(): DEBUG: Fetching logs finished.
    2023/02/13 15:01:57 wazuh-modulesd:gcp-pubsub[10243] wm_gcp.c:143 at wm_gcp_pubsub_main(): DEBUG: Sleeping until: 2023/02/14 15:01:54

</details>

### Conclusions
The module worked as expected, 50 events were sent to `analysisd`.

**Comment by fdalmaup** [comment]
# Azure Test results

The following tests must be performed in sequential order. No data must be modified or removed unless it was specified for the test. 

The tests assume there is a valid authentication file located at `{wazuh_path}/wodles/azure/credentials`.

A test is considered valid if the following conditions are met:

- The expected result described in the test itself is met.
- The expected amount of events were sent to `analysisd`.
- The expected output was present in the `{wazuh_path}/logs/azure-logs.log` file.


## Test results

| Bucket/Service type            | Status                  | Issue ref. |
|--------------------------------|-------------------------|------------|
| Test using ossec.conf in a manager | :green_circle:  | -          |
| Test using ossec.conf in a agent (Python 3.6)   | :green_circle:  | -          |

### Test using ossec.conf in a manager with Python 3.9

Remove the database file and run the module using the following configuration. Check the output and the database status.

<details><summary>Configuration used</summary>

    <wodle name="azure-logs">
	    <disabled>no</disabled>
	    <interval>10m</interval>
	    <run_on_start>yes</run_on_start>

	<graph>
		<auth_path>/var/ossec/wodles/azure/credentials</auth_path>
		<tenantdomain>wazuh.onmicrosoft.com</tenantdomain>
        <request>
            <query>auditLogs/directoryaudits</query>
            <time_offset>30d</time_offset>
        </request>

        <request>
            <query>auditLogs/signIns</query>
            <time_offset>30d</time_offset>
        </request>

        <request>
            <query>auditLogs/provisioning</query>
            <time_offset>30d</time_offset>
        </request>
	</graph>


	<log_analytics>
		<auth_path>/var/ossec/wodles/azure/credentials-analytics</auth_path>
		<tenantdomain>wazuh.onmicrosoft.com</tenantdomain>

        <request>
            <workspace>d6b...efa</workspace>
            <query>AzureActivity</query>
            <time_offset>30d</time_offset>
        </request>
	</log_analytics>

	<storage>
		<auth_path>/var/ossec/wodles/azure/credentials-storage</auth_path>
		<tag>azure-activity</tag>

		<container name="frameworktestcontainer">
			<time_offset>30d</time_offset>
		</container>

		<container name="frameworktestcontainer"></container>

		<container name="frameworktestcontainer"/>
	</storage>
    </wodle>

</details>


<details><summary>Ossec.log output</summary>

    2023/02/13 12:57:09 wazuh-modulesd:azure-logs[18126] wm_azure.c:54 at wm_azure_main(): INFO: Module started.
    2023/02/13 12:57:09 wazuh-modulesd:azure-logs[18126] wm_azure.c:74 at wm_azure_main(): INFO: Starting fetching of logs.
    2023/02/13 12:57:09 wazuh-modulesd:azure-logs[18126] wm_azure.c:85 at wm_azure_main(): INFO: Starting Graphs log collection for the domain 'wazuh.onmicrosoft.com'.
    2023/02/13 12:57:09 wazuh-modulesd:azure-logs[18126] wm_azure.c:209 at wm_azure_graphs(): DEBUG: Creating argument list.
    2023/02/13 12:57:09 wazuh-modulesd:azure-logs[18126] wm_azure.c:259 at wm_azure_graphs(): DEBUG: Launching command: wodles/azure/azure-logs --graph --graph_auth_path /var/ossec/wodles/azure/credentials --graph_tenant_domain wazuh.onmicrosoft.com --graph_tag request_1353914436 --graph_query 'auditLogs/directoryaudits' --graph_time_offset 30d --debug 2
    2023/02/13 12:57:12 wazuh-modulesd:azure-logs[18126] wm_azure.c:276 at wm_azure_graphs(): INFO: Finished Graphs log collection for request 'request_1353914436'.
    2023/02/13 12:57:12 wazuh-modulesd:azure-logs[18126] wm_azure.c:209 at wm_azure_graphs(): DEBUG: Creating argument list.
    2023/02/13 12:57:12 wazuh-modulesd:azure-logs[18126] wm_azure.c:259 at wm_azure_graphs(): DEBUG: Launching command: wodles/azure/azure-logs --graph --graph_auth_path /var/ossec/wodles/azure/credentials --graph_tenant_domain wazuh.onmicrosoft.com --graph_tag request_2106948842 --graph_query 'auditLogs/signIns' --graph_time_offset 30d --debug 2
    2023/02/13 12:57:14 wazuh-modulesd:azure-logs[18126] wm_azure.c:276 at wm_azure_graphs(): INFO: Finished Graphs log collection for request 'request_2106948842'.
    2023/02/13 12:57:14 wazuh-modulesd:azure-logs[18126] wm_azure.c:209 at wm_azure_graphs(): DEBUG: Creating argument list.
    2023/02/13 12:57:14 wazuh-modulesd:azure-logs[18126] wm_azure.c:259 at wm_azure_graphs(): DEBUG: Launching command: wodles/azure/azure-logs --graph --graph_auth_path /var/ossec/wodles/azure/credentials --graph_tenant_domain wazuh.onmicrosoft.com --graph_tag request_847147825 --graph_query 'auditLogs/provisioning' --graph_time_offset 30d --debug 2
    2023/02/13 12:57:16 wazuh-modulesd:azure-logs[18126] wm_azure.c:276 at wm_azure_graphs(): INFO: Finished Graphs log collection for request 'request_847147825'.
    2023/02/13 12:57:16 wazuh-modulesd:azure-logs[18126] wm_azure.c:87 at wm_azure_main(): INFO: Finished Graphs log collection for the domain 'wazuh.onmicrosoft.com'.
    2023/02/13 12:57:16 wazuh-modulesd:azure-logs[18126] wm_azure.c:81 at wm_azure_main(): INFO: Starting Log Analytics collection for the domain 'wazuh.onmicrosoft.com'.
    2023/02/13 12:57:16 wazuh-modulesd:azure-logs[18126] wm_azure.c:120 at wm_azure_log_analytics(): DEBUG: Creating argument list.
    2023/02/13 12:57:16 wazuh-modulesd:azure-logs[18126] wm_azure.c:172 at wm_azure_log_analytics(): DEBUG: Launching command: wodles/azure/azure-logs --log_analytics --la_auth_path /var/ossec/wodles/azure/credentials-analytics --la_tenant_domain wazuh.onmicrosoft.com --la_tag request_996815604 --la_query "AzureActivity" --workspace d6b...efa --la_time_offset 30d --debug 2
    2023/02/13 12:57:18 wazuh-modulesd:azure-logs[18126] wm_azure.c:189 at wm_azure_log_analytics(): INFO: Finished Log Analytics collection for request 'request_996815604'.
    2023/02/13 12:57:18 wazuh-modulesd:azure-logs[18126] wm_azure.c:83 at wm_azure_main(): INFO: Finished Log Analytics collection for the domain 'wazuh.onmicrosoft.com'.
    2023/02/13 12:57:18 wazuh-modulesd:azure-logs[18126] wm_azure.c:92 at wm_azure_main(): INFO: Starting Storage log collection for 'azure-activity'.
    2023/02/13 12:57:18 wazuh-modulesd:azure-logs[18126] wm_azure.c:297 at wm_azure_storage(): DEBUG: Creating argument list.
    2023/02/13 12:57:18 wazuh-modulesd:azure-logs[18126] wm_azure.c:359 at wm_azure_storage(): DEBUG: Launching command: wodles/azure/azure-logs --storage --storage_auth_path /var/ossec/wodles/azure/credentials-storage --container "frameworktestcontainer" --blobs "*" --storage_tag azure-activity --storage_time_offset 30d --debug 2
    2023/02/13 12:57:20 wazuh-modulesd:azure-logs[18126] wm_azure.c:376 at wm_azure_storage(): INFO: Finished Storage log collection for container 'frameworktestcontainer'.
    2023/02/13 12:57:20 wazuh-modulesd:azure-logs[18126] wm_azure.c:297 at wm_azure_storage(): DEBUG: Creating argument list.
    2023/02/13 12:57:20 wazuh-modulesd:azure-logs[18126] wm_azure.c:359 at wm_azure_storage(): DEBUG: Launching command: wodles/azure/azure-logs --storage --storage_auth_path /var/ossec/wodles/azure/credentials-storage --container "frameworktestcontainer" --blobs "*" --storage_tag azure-activity --debug 2
    2023/02/13 12:57:21 wazuh-modulesd:azure-logs[18126] wm_azure.c:376 at wm_azure_storage(): INFO: Finished Storage log collection for container 'frameworktestcontainer'.
    2023/02/13 12:57:21 wazuh-modulesd:azure-logs[18126] wm_azure.c:297 at wm_azure_storage(): DEBUG: Creating argument list.
    2023/02/13 12:57:21 wazuh-modulesd:azure-logs[18126] wm_azure.c:359 at wm_azure_storage(): DEBUG: Launching command: wodles/azure/azure-logs --storage --storage_auth_path /var/ossec/wodles/azure/credentials-storage --container "frameworktestcontainer" --blobs "*" --storage_tag azure-activity --debug 2
    2023/02/13 12:57:23 wazuh-modulesd:azure-logs[18126] wm_azure.c:376 at wm_azure_storage(): INFO: Finished Storage log collection for container 'frameworktestcontainer'.
    2023/02/13 12:57:23 wazuh-modulesd:azure-logs[18126] wm_azure.c:94 at wm_azure_main(): INFO: Finished Storage log collection for 'azure-activity'.
    2023/02/13 12:57:23 wazuh-modulesd:azure-logs[18126] wm_azure.c:100 at wm_azure_main(): DEBUG: Fetching logs finished.
    2023/02/13 12:57:23 wazuh-modulesd:azure-logs[18126] wm_azure.c:70 at wm_azure_main(): DEBUG: Sleeping until: 2023/02/13 13:07:09


</details>


<details><summary>database status after the test</summary>
    
    sqlite3 azure.db 'SELECT * from graph'
    169e36eda7ce0ec7141250c5a2d03285|auditLogs/directoryaudits|2023-01-14T15:57:09.808529Z|2023-02-12T14:50:09.7614201Z
    2012f3812bfc764b6a927f00ddcd7c16|auditLogs/signIns|2023-01-14T15:57:12.905890Z|2023-01-14T15:57:12.905890Z
    6dda2a850e8dae924bc8a63fe246b0ad|auditLogs/provisioning|2023-01-14T15:57:15.459871Z|2023-01-14T15:57:15.459871Z

    sqlite3 azure.db 'SELECT * from log_analytics'
    f90c15b0327962c9e661142547368572|AzureActivity|2023-01-14T15:57:17.357188Z|2023-01-30T23:07:04.860231Z

    sqlite3 azure.db 'SELECT * from storage'
    82e049b81fa6eb88ebf85f1677785f2b|frameworkteststorage|2023-01-14T15:57:20.122713Z|2023-01-14T15:57:20.122713Z

</details>

#### Conclusions

The module worked as expected. The events were sent to the analysis engine.



### Test using ossec.conf in a agent with Python 3.6

Remove the database file and run the module using the following configuration. Check the output and the database status.

<details><summary>Configuration used</summary>

    <wodle name="azure-logs">
        <disabled>no</disabled>
        <interval>10m</interval>
        <run_on_start>yes</run_on_start>

        <graph>
            <auth_path>/var/ossec/wodles/azure/credentials</auth_path>
            <tenantdomain>wazuh.onmicrosoft.com</tenantdomain>
            <request>
                <query>auditLogs/directoryaudits</query>
                <time_offset>30d</time_offset>
            </request>

            <request>
                <query>auditLogs/signIns</query>
                <time_offset>30d</time_offset>
            </request>

            <request>
                <query>auditLogs/provisioning</query>
                <time_offset>30d</time_offset>
            </request>
        </graph>


        <log_analytics>
            <auth_path>/var/ossec/wodles/azure/credentials-analytics</auth_path>
            <tenantdomain>wazuh.onmicrosoft.com</tenantdomain>

            <request>
                <workspace>d6b...efa</workspace>
                <query>AzureActivity</query>
                <time_offset>30d</time_offset>
            </request>
        </log_analytics>

        <storage>
            <auth_path>/var/ossec/wodles/azure/credentials-storage</auth_path>
            <tag>azure-activity</tag>

            <container name="frameworktestcontainer">
                <time_offset>30d</time_offset>
            </container>

            <container name="frameworktestcontainer"></container>

            <container name="frameworktestcontainer"/>
        </storage>
    </wodle>

</details>


<details><summary>Ossec.log output</summary>

    2023/02/13 16:10:09 wazuh-modulesd:azure-logs[12429] wm_azure.c:54 at wm_azure_main(): INFO: Module started.
    2023/02/13 16:10:09 wazuh-modulesd:azure-logs[12429] wm_azure.c:74 at wm_azure_main(): INFO: Starting fetching of logs.
    2023/02/13 16:10:09 wazuh-modulesd:azure-logs[12429] wm_azure.c:85 at wm_azure_main(): INFO: Starting Graphs log collection for the domain 'wazuh.onmicrosoft.com'.
    2023/02/13 16:10:09 wazuh-modulesd:azure-logs[12429] wm_azure.c:209 at wm_azure_graphs(): DEBUG: Creating argument list.
    2023/02/13 16:10:09 wazuh-modulesd:azure-logs[12429] wm_azure.c:259 at wm_azure_graphs(): DEBUG: Launching command: wodles/azure/azure-logs --graph --graph_auth_path /var/ossec/wodles/azure/credentials --graph_tenant_domain wazuh.onmicrosoft.com --graph_tag request_1024477046 --graph_query 'auditLogs/directoryaudits' --graph_time_offset 30d --debug 2
    2023/02/13 16:10:12 wazuh-modulesd:azure-logs[12429] wm_azure.c:276 at wm_azure_graphs(): INFO: Finished Graphs log collection for request 'request_1024477046'.
    2023/02/13 16:10:12 wazuh-modulesd:azure-logs[12429] wm_azure.c:209 at wm_azure_graphs(): DEBUG: Creating argument list.
    2023/02/13 16:10:12 wazuh-modulesd:azure-logs[12429] wm_azure.c:259 at wm_azure_graphs(): DEBUG: Launching command: wodles/azure/azure-logs --graph --graph_auth_path /var/ossec/wodles/azure/credentials --graph_tenant_domain wazuh.onmicrosoft.com --graph_tag request_553957529 --graph_query 'auditLogs/signIns' --graph_time_offset 30d --debug 2
    2023/02/13 16:10:14 wazuh-modulesd:azure-logs[12429] wm_azure.c:276 at wm_azure_graphs(): INFO: Finished Graphs log collection for request 'request_553957529'.
    2023/02/13 16:10:14 wazuh-modulesd:azure-logs[12429] wm_azure.c:209 at wm_azure_graphs(): DEBUG: Creating argument list.
    2023/02/13 16:10:14 wazuh-modulesd:azure-logs[12429] wm_azure.c:259 at wm_azure_graphs(): DEBUG: Launching command: wodles/azure/azure-logs --graph --graph_auth_path /var/ossec/wodles/azure/credentials --graph_tenant_domain wazuh.onmicrosoft.com --graph_tag request_199025250 --graph_query 'auditLogs/provisioning' --graph_time_offset 30d --debug 2
    2023/02/13 16:10:16 wazuh-modulesd:azure-logs[12429] wm_azure.c:276 at wm_azure_graphs(): INFO: Finished Graphs log collection for request 'request_199025250'.
    2023/02/13 16:10:16 wazuh-modulesd:azure-logs[12429] wm_azure.c:87 at wm_azure_main(): INFO: Finished Graphs log collection for the domain 'wazuh.onmicrosoft.com'.
    2023/02/13 16:10:16 wazuh-modulesd:azure-logs[12429] wm_azure.c:81 at wm_azure_main(): INFO: Starting Log Analytics collection for the domain 'wazuh.onmicrosoft.com'.
    2023/02/13 16:10:16 wazuh-modulesd:azure-logs[12429] wm_azure.c:120 at wm_azure_log_analytics(): DEBUG: Creating argument list.
    2023/02/13 16:10:16 wazuh-modulesd:azure-logs[12429] wm_azure.c:172 at wm_azure_log_analytics(): DEBUG: Launching command: wodles/azure/azure-logs --log_analytics --la_auth_path /var/ossec/wodles/azure/credentials-analytics --la_tenant_domain wazuh.onmicrosoft.com --la_tag request_345878887 --la_query "AzureActivity" --workspace d6b...efa --la_time_offset 30d --debug 2
    2023/02/13 16:10:19 wazuh-modulesd:azure-logs[12429] wm_azure.c:189 at wm_azure_log_analytics(): INFO: Finished Log Analytics collection for request 'request_345878887'.
    2023/02/13 16:10:19 wazuh-modulesd:azure-logs[12429] wm_azure.c:83 at wm_azure_main(): INFO: Finished Log Analytics collection for the domain 'wazuh.onmicrosoft.com'.
    2023/02/13 16:10:19 wazuh-modulesd:azure-logs[12429] wm_azure.c:92 at wm_azure_main(): INFO: Starting Storage log collection for 'azure-activity'.
    2023/02/13 16:10:19 wazuh-modulesd:azure-logs[12429] wm_azure.c:297 at wm_azure_storage(): DEBUG: Creating argument list.
    2023/02/13 16:10:19 wazuh-modulesd:azure-logs[12429] wm_azure.c:359 at wm_azure_storage(): DEBUG: Launching command: wodles/azure/azure-logs --storage --storage_auth_path /var/ossec/wodles/azure/credentials-storage --container "frameworktestcontainer" --blobs "*" --storage_tag azure-activity --storage_time_offset 30d --debug 2
    2023/02/13 16:10:20 wazuh-modulesd:azure-logs[12429] wm_azure.c:376 at wm_azure_storage(): INFO: Finished Storage log collection for container 'frameworktestcontainer'.
    2023/02/13 16:10:20 wazuh-modulesd:azure-logs[12429] wm_azure.c:297 at wm_azure_storage(): DEBUG: Creating argument list.
    2023/02/13 16:10:20 wazuh-modulesd:azure-logs[12429] wm_azure.c:359 at wm_azure_storage(): DEBUG: Launching command: wodles/azure/azure-logs --storage --storage_auth_path /var/ossec/wodles/azure/credentials-storage --container "frameworktestcontainer" --blobs "*" --storage_tag azure-activity --debug 2
    2023/02/13 16:10:22 wazuh-modulesd:azure-logs[12429] wm_azure.c:376 at wm_azure_storage(): INFO: Finished Storage log collection for container 'frameworktestcontainer'.
    2023/02/13 16:10:22 wazuh-modulesd:azure-logs[12429] wm_azure.c:297 at wm_azure_storage(): DEBUG: Creating argument list.
    2023/02/13 16:10:22 wazuh-modulesd:azure-logs[12429] wm_azure.c:359 at wm_azure_storage(): DEBUG: Launching command: wodles/azure/azure-logs --storage --storage_auth_path /var/ossec/wodles/azure/credentials-storage --container "frameworktestcontainer" --blobs "*" --storage_tag azure-activity --debug 2
    2023/02/13 16:10:24 wazuh-modulesd:azure-logs[12429] wm_azure.c:376 at wm_azure_storage(): INFO: Finished Storage log collection for container 'frameworktestcontainer'.
    2023/02/13 16:10:24 wazuh-modulesd:azure-logs[12429] wm_azure.c:94 at wm_azure_main(): INFO: Finished Storage log collection for 'azure-activity'.
    2023/02/13 16:10:24 wazuh-modulesd:azure-logs[12429] wm_azure.c:100 at wm_azure_main(): DEBUG: Fetching logs finished.
    2023/02/13 16:10:24 wazuh-modulesd:azure-logs[12429] wm_azure.c:70 at wm_azure_main(): DEBUG: Sleeping until: 2023/02/13 16:20:09

</details>


<details><summary>database status after the test</summary>

    sqlite3 azure.db 'SELECT * from graph'
    169e36eda7ce0ec7141250c5a2d03285|auditLogs/directoryaudits|2023-01-14T16:10:10.409625Z|2023-02-12T14:50:09.7614201Z
    2012f3812bfc764b6a927f00ddcd7c16|auditLogs/signIns|2023-01-14T16:10:12.898297Z|2023-01-14T16:10:12.898297Z
    6dda2a850e8dae924bc8a63fe246b0ad|auditLogs/provisioning|2023-01-14T16:10:14.950398Z|2023-01-14T16:10:14.950398Z

    sqlite3 azure.db 'SELECT * from log_analytics'
    f90c15b0327962c9e661142547368572|AzureActivity|2023-01-14T16:10:17.031665Z|2023-01-30T23:07:04.860231Z

    sqlite3 azure.db 'SELECT * from storage'
    82e049b81fa6eb88ebf85f1677785f2b|frameworkteststorage|2023-01-14T16:10:20.262532Z|2023-01-14T16:10:20.262532Z

</details>

#### Conclusions

The module worked as expected. The events were sent to the analysis engine.

**Comment by fdalmaup** [comment]
# Docker Listener Test results


The [logall_json option](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#logall_json) must be enabled in the `ossec.conf` file before starting the tests.

## Test docker listener using ossec.conf in a manager

Apply the following configuration, restart Wazuh and check ossec.log to verify the module is running
```
<wodle name="docker-listener">
    <interval>30s</interval>
    <attempts>5</attempts>
    <run_on_start>yes</run_on_start>
    <disabled>no</disabled>
</wodle>
```

<details><summary>ossec.log output</summary>

    2023/02/13 13:22:27 wazuh-modulesd[20840] main.c:95 at main(): DEBUG: Created new thread for the 'docker-listener' module.
    2023/02/13 13:22:27 wazuh-modulesd:docker-listener[20840] wm_docker.c:46 at wm_docker_main(): INFO: Module docker-listener started.
    2023/02/13 13:22:27 wazuh-modulesd:docker-listener[20840] wm_docker.c:59 at wm_docker_main(): INFO: Starting to listening Docker events.
    2023/02/13 13:22:27 wazuh-modulesd:docker-listener[20840] wm_docker.c:63 at wm_docker_main(): DEBUG: Launching command 'wodles/docker/DockerListener'

</details>


Run a container with no cache available
```
docker run ubuntu
```

<details><summary>Archives.json output</summary>


    {"timestamp":"2023-02-13T16:27:53.941+0000","rule":{"level":3,"description":"Docker: Image or repository ubuntu pulled","id":"87932","firedtimes":1,"mail":false,"groups":["docker"],"pci_dss":["10.2.7"],"hipaa":["164.312.b"],"nist_800_53":["AU.14"],"tsc":["CC6.8","CC7.2","CC7.3"]},"agent":{"id":"000","name":"94ab9303e773"},"manager":{"name":"94ab9303e773"},"id":"1676305673.1862519","cluster":{"name":"wazuh","node":"manager"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"pull\", \"id\": \"ubuntu:latest\", \"Type\": \"image\", \"Action\": \"pull\", \"Actor\": {\"ID\": \"ubuntu:latest\", \"Attributes\": {\"name\": \"ubuntu\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1676305673, \"timeNano\": 1676305673941344388}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"pull","id":"ubuntu:latest","Type":"image","Action":"pull","Actor":{"ID":"ubuntu:latest","Attributes":{"name":"ubuntu","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1676305673","timeNano":"1676305673941344512.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-02-13T16:27:54.023+0000","rule":{"level":3,"description":"Docker: Container pedantic_elion created","id":"87901","firedtimes":2,"mail":false,"groups":["docker"],"pci_dss":["10.2.7"],"hipaa":["164.312.b"],"nist_800_53":["AU.14"],"tsc":["CC6.8","CC7.2","CC7.3"]},"agent":{"id":"000","name":"94ab9303e773"},"manager":{"name":"94ab9303e773"},"id":"1676305674.1863490","cluster":{"name":"wazuh","node":"manager"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"create\", \"id\": \"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"create\", \"Actor\": {\"ID\": \"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"pedantic_elion\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1676305674, \"timeNano\": 1676305674022908662}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"create","id":"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0","from":"ubuntu","Type":"container","Action":"create","Actor":{"ID":"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0","Attributes":{"image":"ubuntu","name":"pedantic_elion","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1676305674","timeNano":"1676305674022908672.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-02-13T16:27:54.024+0000","rule":{"level":3,"description":"Docker: Attached local standard input, output, and error streams to container pedantic_elion","id":"87922","firedtimes":2,"mail":false,"groups":["docker"]},"agent":{"id":"000","name":"94ab9303e773"},"manager":{"name":"94ab9303e773"},"id":"1676305674.1864791","cluster":{"name":"wazuh","node":"manager"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"attach\", \"id\": \"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"attach\", \"Actor\": {\"ID\": \"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"pedantic_elion\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1676305674, \"timeNano\": 1676305674023880239}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"attach","id":"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0","from":"ubuntu","Type":"container","Action":"attach","Actor":{"ID":"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0","Attributes":{"image":"ubuntu","name":"pedantic_elion","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1676305674","timeNano":"1676305674023880192.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-02-13T16:27:54.035+0000","rule":{"level":3,"description":"Docker: Network bridge connected","id":"87928","firedtimes":2,"mail":false,"groups":["docker"]},"agent":{"id":"000","name":"94ab9303e773"},"manager":{"name":"94ab9303e773"},"id":"1676305674.1866065","cluster":{"name":"wazuh","node":"manager"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"Type\": \"network\", \"Action\": \"connect\", \"Actor\": {\"ID\": \"98142bb3e0328d83b031db00ca67c5d3f55044bfcabdb7a7fd89e36cec4cb9c8\", \"Attributes\": {\"container\": \"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0\", \"name\": \"bridge\", \"type\": \"bridge\"}}, \"scope\": \"local\", \"time\": 1676305674, \"timeNano\": 1676305674035184933}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"Type":"network","Action":"connect","Actor":{"ID":"98142bb3e0328d83b031db00ca67c5d3f55044bfcabdb7a7fd89e36cec4cb9c8","Attributes":{"container":"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0","name":"bridge","type":"bridge"}},"scope":"local","time":"1676305674","timeNano":"1676305674035184896.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-02-13T16:27:54.301+0000","rule":{"level":3,"description":"Docker: Container pedantic_elion started","id":"87903","firedtimes":2,"mail":false,"groups":["docker"]},"agent":{"id":"000","name":"94ab9303e773"},"manager":{"name":"94ab9303e773"},"id":"1676305674.1866988","cluster":{"name":"wazuh","node":"manager"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"start\", \"id\": \"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"start\", \"Actor\": {\"ID\": \"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"pedantic_elion\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1676305674, \"timeNano\": 1676305674301038466}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"start","id":"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0","from":"ubuntu","Type":"container","Action":"start","Actor":{"ID":"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0","Attributes":{"image":"ubuntu","name":"pedantic_elion","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1676305674","timeNano":"1676305674301038592.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-02-13T16:27:54.469+0000","rule":{"level":4,"description":"Docker: Network bridge disconnected","id":"87929","firedtimes":2,"mail":false,"groups":["docker"]},"agent":{"id":"000","name":"94ab9303e773"},"manager":{"name":"94ab9303e773"},"id":"1676305674.1868206","cluster":{"name":"wazuh","node":"manager"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"Type\": \"network\", \"Action\": \"disconnect\", \"Actor\": {\"ID\": \"98142bb3e0328d83b031db00ca67c5d3f55044bfcabdb7a7fd89e36cec4cb9c8\", \"Attributes\": {\"container\": \"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0\", \"name\": \"bridge\", \"type\": \"bridge\"}}, \"scope\": \"local\", \"time\": 1676305674, \"timeNano\": 1676305674468669689}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"Type":"network","Action":"disconnect","Actor":{"ID":"98142bb3e0328d83b031db00ca67c5d3f55044bfcabdb7a7fd89e36cec4cb9c8","Attributes":{"container":"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0","name":"bridge","type":"bridge"}},"scope":"local","time":"1676305674","timeNano":"1676305674468669696.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-02-13T16:27:54.481+0000","rule":{"level":7,"description":"Docker: Container pedantic_elion received the action: die","id":"87924","firedtimes":2,"mail":false,"groups":["docker"],"gdpr":["IV_32.2"]},"agent":{"id":"000","name":"94ab9303e773"},"manager":{"name":"94ab9303e773"},"id":"1676305674.1869138","cluster":{"name":"wazuh","node":"manager"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"die\", \"id\": \"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"die\", \"Actor\": {\"ID\": \"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0\", \"Attributes\": {\"exitCode\": \"0\", \"image\": \"ubuntu\", \"name\": \"pedantic_elion\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1676305674, \"timeNano\": 1676305674480608767}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"die","id":"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0","from":"ubuntu","Type":"container","Action":"die","Actor":{"ID":"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0","Attributes":{"exitCode":"0","image":"ubuntu","name":"pedantic_elion","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1676305674","timeNano":"1676305674480608768.000000"}},"location":"Wazuh-Docker"}

</details>


Remove the image

<details><summary>Archives.json output</summary>

    {"timestamp":"2023-02-13T16:29:17.129+0000","rule":{"level":5,"description":"Docker: Container pedantic_elion destroyed","id":"87902","mitre":{"id":["T1561.001"],"tactic":["Impact"],"technique":["Disk Content Wipe"]},"firedtimes":2,"mail":false,"groups":["docker"],"pci_dss":["10.2.7","11.5"],"hipaa":["164.312.b","164.312.c.1","164.312.c.2"],"nist_800_53":["AU.14","SI.7"],"tsc":["CC6.8","CC7.2","CC7.3","PI1.4","PI1.5","CC6.1"]},"agent":{"id":"000","name":"94ab9303e773"},"manager":{"name":"94ab9303e773"},"id":"1676305757.1870431","cluster":{"name":"wazuh","node":"manager"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"destroy\", \"id\": \"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0\", \"from\": \"ubuntu\", \"Type\": \"container\", \"Action\": \"destroy\", \"Actor\": {\"ID\": \"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0\", \"Attributes\": {\"image\": \"ubuntu\", \"name\": \"pedantic_elion\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1676305757, \"timeNano\": 1676305757129150880}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"destroy","id":"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0","from":"ubuntu","Type":"container","Action":"destroy","Actor":{"ID":"4c2ef8f756c667b1f1c4a86faa499f558189b0af8ec690b35e54f63ca25a33f0","Attributes":{"image":"ubuntu","name":"pedantic_elion","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1676305757","timeNano":"1676305757129150976.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-02-13T16:29:18.985+0000","rule":{"level":3,"description":"Docker: Image sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02 untagged","id":"87919","firedtimes":2,"mail":false,"groups":["docker"]},"agent":{"id":"000","name":"94ab9303e773"},"manager":{"name":"94ab9303e773"},"id":"1676305758.1871834","cluster":{"name":"wazuh","node":"manager"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"untag\", \"id\": \"sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02\", \"Type\": \"image\", \"Action\": \"untag\", \"Actor\": {\"ID\": \"sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02\", \"Attributes\": {\"name\": \"sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02\", \"org.opencontainers.image.ref.name\": \"ubuntu\", \"org.opencontainers.image.version\": \"22.04\"}}, \"scope\": \"local\", \"time\": 1676305758, \"timeNano\": 1676305758984705579}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"untag","id":"sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02","Type":"image","Action":"untag","Actor":{"ID":"sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02","Attributes":{"name":"sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02","org":{"opencontainers":{"image":{"ref":{"name":"ubuntu"},"version":"22.04"}}}}},"scope":"local","time":"1676305758","timeNano":"1676305758984705536.000000"}},"location":"Wazuh-Docker"}
    {"timestamp":"2023-02-13T16:29:19.021+0000","rule":{"level":7,"description":"Docker: Container sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02 deleted","id":"87921","mitre":{"id":["T1070.004","T1561.001"],"tactic":["Defense Evasion","Impact"],"technique":["File Deletion","Disk Content Wipe"]},"firedtimes":2,"mail":false,"groups":["docker"],"pci_dss":["10.2.7","11.5"],"hipaa":["164.312.b","164.312.c.1","164.312.c.2"],"nist_800_53":["AU.14","SI.7"],"tsc":["CC6.8","CC7.2","CC7.3","PI1.4","PI1.5","CC6.1"]},"agent":{"id":"000","name":"94ab9303e773"},"manager":{"name":"94ab9303e773"},"id":"1676305759.1873145","cluster":{"name":"wazuh","node":"manager"},"full_log":"{\"integration\": \"docker\", \"docker\": {\"status\": \"delete\", \"id\": \"sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02\", \"Type\": \"image\", \"Action\": \"delete\", \"Actor\": {\"ID\": \"sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02\", \"Attributes\": {\"name\": \"sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02\"}}, \"scope\": \"local\", \"time\": 1676305759, \"timeNano\": 1676305759020984661}}","decoder":{"name":"json"},"data":{"integration":"docker","docker":{"status":"delete","id":"sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02","Type":"image","Action":"delete","Actor":{"ID":"sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02","Attributes":{"name":"sha256:58db3edaf2be6e80f628796355b1bdeaf8bea1692b402f48b7e7b8d1ff100b02"}},"scope":"local","time":"1676305759","timeNano":"1676305759020984576.000000"}},"location":"Wazuh-Docker"}

</details>


### Conclusions

The module worked as expected

**Comment by fdalmaup** [comment]
# Shuffle Test Results

The test assumes there is a functional Shuffle instance with an available webhook.

A test is considered valid if the following conditions are met:

- The expected result described in the test itself is met.
- The expected output was present in the `{wazuh_path}/logs/integrations.log` and `{wazuh_path}/logs/ossec.log` file.


## Test using ossec.conf in a manager

The following configuration has to be added to `ossec.conf`
```
<integration>
     <name>shuffle</name>
     <hook_url>http://<IP>:3001/api/v1/hooks/<HOOK_ID></hook_url>
     <level>3</level>
     <alert_format>json</alert_format>
 </integration>
```
Also, the configuration `integrator.debug=2` to `local_internal_options.conf `.

Restart the Wazuh Manager.

<details><summary>ossec.log</summary>
    
    2023/02/13 18:20:29 wazuh-integratord[16728] integrator.c:403 at OS_IntegratorD(): DEBUG: Running: integrations /tmp/shuffle-1676312429--440813932.alert  http://<IP>:3001/api/v1/hooks/<HOOK_ID> debug
    2023/02/13 18:20:30 wazuh-integratord[16728] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: /var/ossec/framework/python/lib/python3.9/site-packages/urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host '172.21.0.6'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
    2023/02/13 18:20:30 wazuh-integratord[16728] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Mon Feb 13 18:20:30 UTC 2023: # Starting
    2023/02/13 18:20:30 wazuh-integratord[16728] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Mon Feb 13 18:20:30 UTC 2023: # Webhook
    2023/02/13 18:20:30 wazuh-integratord[16728] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Mon Feb 13 18:20:30 UTC 2023: http://<IP>:3001/api/v1/hooks/<HOOK_ID>
    2023/02/13 18:20:30 wazuh-integratord[16728] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Mon Feb 13 18:20:30 UTC 2023: # File location
    2023/02/13 18:20:30 wazuh-integratord[16728] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Mon Feb 13 18:20:30 UTC 2023: /tmp/shuffle-1676312429--440813932.alert
    2023/02/13 18:20:30 wazuh-integratord[16728] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Mon Feb 13 18:20:30 UTC 2023: # Processing alert
    2023/02/13 18:20:30 wazuh-integratord[16728] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Mon Feb 13 18:20:30 UTC 2023: {'timestamp': '2023-02-13T18:20:28.152+0000', 'rule': {'level': 3, 'description': 'Ossec server started.', 'id': '502', 'firedtimes': 1, 'mail': False, 'groups': ['ossec'], 'pci_dss': ['10.6.1'], 'gpg13': ['10.1'], 'gdpr': ['IV_35.7.d'], 'hipaa': ['164.312.b'], 'nist_800_53': ['AU.6'], 'tsc': ['CC7.2', 'CC7.3']}, 'agent': {'id': '000', 'name': '547658cb07c1'}, 'manager': {'name': '547658cb07c1'}, 'id': '1676312428.510102', 'full_log': 'ossec: Manager started.', 'decoder': {'name': 'ossec'}, 'location': 'wazuh-monitord'}
    2023/02/13 18:20:30 wazuh-integratord[16728] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Mon Feb 13 18:20:30 UTC 2023: # Generating message
    2023/02/13 18:20:30 wazuh-integratord[16728] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Mon Feb 13 18:20:30 UTC 2023: {"severity": 1, "pretext": "WAZUH Alert", "title": "Ossec server started.", "text": "ossec: Manager started.", "rule_id": "502", "timestamp": "2023-02-13T18:20:28.152+0000", "id": "1676312428.510102", "all_fields": {"timestamp": "2023-02-13T18:20:28.152+0000", "rule": {"level": 3, "description": "Ossec server started.", "id": "502", "firedtimes": 1, "mail": false, "groups": ["ossec"], "pci_dss": ["10.6.1"], "gpg13": ["10.1"], "gdpr": ["IV_35.7.d"], "hipaa": ["164.312.b"], "nist_800_53": ["AU.6"], "tsc": ["CC7.2", "CC7.3"]}, "agent": {"id": "000", "name": "547658cb07c1"}, "manager": {"name": "547658cb07c1"}, "id": "1676312428.510102", "full_log": "ossec: Manager started.", "decoder": {"name": "ossec"}, "location": "wazuh-monitord"}}
    2023/02/13 18:20:30 wazuh-integratord[16728] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Mon Feb 13 18:20:30 UTC 2023: # Sending message
    2023/02/13 18:20:30 wazuh-integratord[16728] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Mon Feb 13 18:20:30 UTC 2023: # In send msg
    2023/02/13 18:20:30 wazuh-integratord[16728] integrator.c:412 at OS_IntegratorD(): DEBUG: integratord: Mon Feb 13 18:20:30 UTC 2023: # After send msg: <Response [200]>

</details>

<details><summary>integrations.log</summary>

    Mon Feb 13 18:20:30 UTC 2023 /tmp/shuffle-1676312429--440813932.alert  http://<IP>:3001/api/v1/hooks/<HOOK_ID> debug
    Mon Feb 13 18:20:30 UTC 2023: # Starting
    Mon Feb 13 18:20:30 UTC 2023: # Webhook
    Mon Feb 13 18:20:30 UTC 2023: http://<IP>:3001/api/v1/hooks/<HOOK_ID>
    Mon Feb 13 18:20:30 UTC 2023: # File location
    Mon Feb 13 18:20:30 UTC 2023: /tmp/shuffle-1676312429--440813932.alert
    Mon Feb 13 18:20:30 UTC 2023: # Processing alert
    Mon Feb 13 18:20:30 UTC 2023: {'timestamp': '2023-02-13T18:20:28.152+0000', 'rule': {'level': 3, 'description': 'Ossec server started.', 'id': '502', 'firedtimes': 1, 'mail': False, 'groups': ['ossec'], 'pci_dss': ['10.6.1'], 'gpg13': ['10.1'], 'gdpr': ['IV_35.7.d'], 'hipaa': ['164.312.b'], 'nist_800_53': ['AU.6'], 'tsc': ['CC7.2', 'CC7.3']}, 'agent': {'id': '000', 'name': '547658cb07c1'}, 'manager': {'name': '547658cb07c1'}, 'id': '1676312428.510102', 'full_log': 'ossec: Manager started.', 'decoder': {'name': 'ossec'}, 'location': 'wazuh-monitord'}
    Mon Feb 13 18:20:30 UTC 2023: # Generating message
    Mon Feb 13 18:20:30 UTC 2023: {"severity": 1, "pretext": "WAZUH Alert", "title": "Ossec server started.", "text": "ossec: Manager started.", "rule_id": "502", "timestamp": "2023-02-13T18:20:28.152+0000", "id": "1676312428.510102", "all_fields": {"timestamp": "2023-02-13T18:20:28.152+0000", "rule": {"level": 3, "description": "Ossec server started.", "id": "502", "firedtimes": 1, "mail": false, "groups": ["ossec"], "pci_dss": ["10.6.1"], "gpg13": ["10.1"], "gdpr": ["IV_35.7.d"], "hipaa": ["164.312.b"], "nist_800_53": ["AU.6"], "tsc": ["CC7.2", "CC7.3"]}, "agent": {"id": "000", "name": "547658cb07c1"}, "manager": {"name": "547658cb07c1"}, "id": "1676312428.510102", "full_log": "ossec: Manager started.", "decoder": {"name": "ossec"}, "location": "wazuh-monitord"}}
    Mon Feb 13 18:20:30 UTC 2023: # Sending message
    Mon Feb 13 18:20:30 UTC 2023: # In send msg
    Mon Feb 13 18:20:30 UTC 2023: # After send msg: <Response [200]>

</details>

### Conclusions
The integration works as expected, sending the alerts to Shuffle.
